frontend/src/metabase/lib/dom.js

```diff
@@ -350,12 +367,39 @@ const getLocation = url => {
   try {
     const { pathname, search, hash } = new URL(url, window.location.origin);
     const query = querystring.parse(search.substring(1));
-    return { pathname, search, query, hash };
+    return {
+      pathname: getPathnameWithoutSubPath(pathname),
+      search,
+      query,
+      hash,
+    };
   } catch {
     return {};
   }
 };
 
+function getPathnameWithoutSubPath(pathname) {
```

Comment:
@ranquild I didn't go with the approach of replacing pathname with site path because `pathname` could have different cases than the site path e.g. if we have `site-url` as `https://mysite.com/METABASE/`, but visit the URL `https://mysite.com/metabase/question/1`. We need to be able to match that the visited URL is a Metabase URL